src/Smalot/PdfParser/RawData/RawDataParser.php

```diff
@@ -762,17 +755,18 @@ protected function getRawObject(string $pdfData, int $offset = 0): array
                     // start stream object
                     $objtype = 'stream';
                     $offset += 6;
-                    if (1 == preg_match('/^([\r]?[\n])/isU', substr($pdfData, $offset), $matches)) {
+                    if (1 == preg_match('/^([\r]?[\n])/isU', substr($pdfData, $offset, 4), $matches)) {
                         $offset += \strlen($matches[0]);
```

Comment:
if this stream object appears in the beginning of 20M pdf this unlimited substr results in almost 20M RAM allocation.

While this regexp can match only 1 or 2 character string